modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1241 @@
+    is.character(out_get_data)
+  )
+  if (!is.null(out_process_data)) {
+    PEcAn.logger::severeifnot("out_process_data should be of character type",
+                              is.character(out_process_data))
+  }
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  # extract the AOI of the site from BETYdb
+  coords <-
+    unlist(PEcAn.DB::db.query(
+      sprintf("select ST_AsGeoJSON(geometry) from sites where id=%f", siteid),
+      con = dbcon
+    ), use.names = FALSE)
+  
+  if(!(tolower(gsub(".*type(.+),coordinates.*", "\\1",  gsub("[^=A-Za-z,0-9{} ]+","",coords))) %in% reg_info$coordtype)){
+    PEcAn.logger::logger.severe(paste0("Coordinate type of the site is not supported by the requested source, please make sure that your site type is ", reg_info$coordtype))
+  }
+  
+  # construct raw file name
+  remotedata_file_names <- construct_remotedata_filename(source, collection, siteid_short, scale, projection, qc, algorithm, out_process_data)
+  
+  raw_file_name <- remotedata_file_names$raw_file_name
+  
+  pro_file_name <- remotedata_file_names$pro_file_name
+
+  
+  # check if any data is already present in the inputs table
+  dbstatus <-
+    remotedata_db_check(
+      raw_file_name     = raw_file_name,
+      pro_file_name     = pro_file_name,
+      start             = start,
+      end               = end,
+      siteid            = siteid,
+      siteid_short      = siteid_short,
+      out_get_data      = out_get_data,
+      algorithm         = algorithm,
+      out_process_data  = out_process_data,
+      overwrite         = overwrite,
+      dbcon             = dbcon
+    )
+  
+  remotefile_check_flag  <- dbstatus$remotefile_check_flag 
+  start                  <- dbstatus$start                 
+  end                    <- dbstatus$end                   
+  stage_get_data         <- dbstatus$stage_get_data        
+  write_raw_start        <- dbstatus$write_raw_start       
+  write_raw_end          <- dbstatus$write_raw_end         
+  raw_merge              <- dbstatus$raw_merge             
+  existing_raw_file_path <- dbstatus$existing_raw_file_path
+  stage_process_data     <- dbstatus$stage_process_data    
+  write_pro_start        <- dbstatus$write_pro_start       
+  write_pro_end          <- dbstatus$write_pro_end         
+  pro_merge              <- dbstatus$pro_merge             
+  input_file             <- dbstatus$input_file            
+  existing_pro_file_path <- dbstatus$existing_pro_file_path
+  raw_check              <- dbstatus$raw_check             
+  pro_check              <- dbstatus$pro_check
+  
+  
+  
+  if(stage_get_data == FALSE && stage_process_data == FALSE){
+    # requested data already exists, no need to call rp_control
+    
+    pro_id   <- pro_check$id
+    pro_path <- pro_check$file_path
+    raw_id   <- raw_check$id
+    raw_path <- raw_check$file_path
+  
+    settings$remotedata$raw_id   <- raw_id
```

Comment:
```suggestion
    settings$remotedata$raw_id   <-  raw_check$id
```
Is there a reason not to be more concise and directly pass the ids and paths like this and get rid of the 4 extra lines above?